torch/jit/frontend.py

```diff
@@ -363,6 +378,94 @@ def build_param(ctx, py_arg, self_name, kwarg_only):
+        # with their annotated types
+        inputs = []
+        outputs = []
+        for arg in args:
+            var_name = arg.arg
+            var_ann = arg.value.value
+            var_decl_type, var_ann = var_ann.split(":")
+            if var_decl_type == "inp":
+                inputs.append((var_name, var_ann))
+            if var_decl_type == "out":
+                outputs.append((var_name, var_ann))
+        return inputs, outputs
+
+    def create_unique_name_ext(ctx, stmt):
+        # extension will be based on the filename plus the line number
+        # of original context manager
+        return os.path.basename(ctx.filename).replace(".", "_") + "_" + str(stmt.lineno)
+
+    def build_return_ann_stmt(outputs):
+        return_type_ann = ""
+        return_statement_str = "return "
+        if len(outputs) == 0:
+            return_type_ann += " -> None: pass"
+        if len(outputs) == 1:
+            return_type_ann = " -> " + outputs[0][1] + ": pass"
+            return_statement_str += outputs[0][0]
+        if len(outputs) > 1:
+            return_type_ann = "Tuple["
+            for var in outputs:
+                var_name, var_ann = var
+                return_type_ann += var_ann + ", "
+                return_statement_str += var_name + ", "
+            return_type_ann = " -> " + return_type_ann[:-2] + "]" + ": pass"
+            return_statement_str = return_statement_str[:-2]
+        return return_type_ann, return_statement_str
+
+    def build_args(args):
+        result = ""
+        for arg in args:
+            var_name, _ = arg
+            result += var_name + ", "
+        return result[:-2]
+
+    inputs, outputs = process_ins_outs(stmt.items[0].context_expr.keywords)
+
+    # build the replacement function str with given inputs and outputs
+    ignore_function_name = "func_ignore_" + create_unique_name_ext(ctx, stmt)
+    ignore_function_str = "\ndef " + ignore_function_name + "("
+    for var in inputs:
+        var_name, var_ann = var
+        ignore_function_str += var_name + " :" + var_ann + ", "
+    ignore_function_str = ignore_function_str[:-2] + ")"
+
+    return_ann, return_stmt = build_return_ann_stmt(outputs)
+    ignore_function_str += return_ann
+
+    # first create the functionDef object from just declaration
+    ignore_function = ast.parse(ignore_function_str).body[0]
+
+    # dump the body of context manager to dummy function
+    ignore_function.body = stmt.body
+
+    # insert return statement to the function
+    return_stmt = ast.parse(return_stmt).body[0]
+    ignore_function.body.append(return_stmt)
+
+    # registers the custom function in the global context
+    ignore_func_str = ast.unparse(ignore_function)
+    ignore_func_str += "\nglobals()[\"{}\"] = {}".format(ignore_function_name, ignore_function_name)
+    ignore_func_str += "\nglobals()[\"{}\"]._torchscript_modifier = FunctionModifiers.IGNORE".format(ignore_function_name)
```

Comment:
Maybe better to just generate a call to `torch.jit.ignore` decorator. That would make sure future changes we make to `torch.jit.ignore` automatically applies to functions we generate.

Reply:
Oh that is a good idea.